pulp_ansible/app/tasks/collections.py

```diff
@@ -81,6 +81,26 @@ def sync(remote_pk, repository_pk, mirror):
     d_version = AnsibleDeclarativeVersion(first_stage, repository, mirror=mirror)
     d_version.create()
 
+    repo_version = repository.latest_version()
+    collections = Collection.objects.filter(versions__in=repo_version.content).distinct()
+    to_deprecate = []
+    to_undeprecate = []
+    for collection in collections:
```

Comment:
I had imagined this part differently. Instead of iterating through the db's collection's you could filter them with a `Q(namespace==..., name=...)` and || those all together to just get the subset of collections that need to have `AnsibleCollectionDeprecated` objects created, then use the bulk_create with ignore_conflicts to create them.

For deletion I was thinking take the set of collections used in the creation, say that is `collections_deprecate_true_qs` then do something like `AnsiblecollectionDeprecated.objects.exclude(collection__pk__in=collections_deprecate_true_qs).delete()`.

Reply:
I tested locally, doing:
```python
AnsiblecollectionDeprecated.objects.exclude(collection__pk__in=collections_deprecate_true_qs).delete()
```
would not cover the case where no collection was deprecated and a collection from the previous sync was undeprecated

Reply:
I suspect you're onto something, but can you help me understand the case a bit more? I don't follow.